### ISSUE TITLE
Failed to build in Cygwin

### ISSUE THREAD

**Issue by yjfvictor** [tdlib/td#529]
I tried to build the code on Cygwin 3.0.5, with gcc 7.4.0, and it reports the following errors:

[cygwin_make.txt](https://github.com/tdlib/td/files/3036626/make.txt)

**Comment by levlam** [comment]
Remove in file `sqlite/sqlite/sqlite3.c` at line 13415 `|| defined(__CYGWIN__)`. It's odd that SQLite thinks that Cygwin is a Windows environment.

**Comment by levlam** [comment]
Also, you will need to build in Release mode: `cmake -DCMAKE_BUILD_TYPE=Release ..`.

**Comment by levlam** [comment]
@yjfvictor I've just succesfully built TDLib 1.3.0 in Release mode on Cygwin 3.0.5 with gcc 7.4.0. Have you also succeeded?

**Comment by levlam** [comment]
Investigating further, it is a bug https://github.com/sqlcipher/sqlcipher/issues/206 in SQLCipher amalgamation. The amalgamation has commented include
```
/* #include "windows.h" */
```
on line 27197 because of include
```
# include <windows.h>
```
on line 18748, but the latter is only conditionally included under Windows and not under Cygwin. So, windows.h doesn't included at all on Cygwin.

It seems that the better way to fix compilation error is to include the header by adding
```
#define NOCRYPT
#include <windows.h>
```
near line 27197 (or at the beginning of the `sqlite/sqlite/sqlite3.c` file).

**Comment by yjfvictor** [comment]
@levlam 
I still failed to build it on Cygwin 3.0.5, with gcc 7.4.0

[build_errors.txt](https://github.com/tdlib/td/files/3053407/build_errors.txt)

**Comment by yjfvictor** [comment]
OK, fine. I've built it successfully.

**Comment by levlam** [comment]
We have patched the SQLCipher in TDLib 1.4.0 to avoid this compilation error.
